packages/workflow/src/Extensions/ArrayExtensions.ts

```diff
@@ -281,28 +211,57 @@ function renameKeys(value: unknown[], extraArgs: string[]): unknown[] {
 	});
 }
 
-function merge(value: unknown[], extraArgs: unknown[][]): unknown[] {
+function mergeObjects(value: object, extraArgs: unknown[]): unknown {
+	const [other] = extraArgs;
+
+	if (!other) {
+		return value;
+	}
+
+	if (typeof other !== 'object') {
+		throw new ExpressionExtensionError('merge(): expected object arg');
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
We can remove both of these comments if you assert that the type of `value`  is `Record<string, unknown>` instead of a general `object`. This will allow us to use the spread operator without the warning. And since we're asserting it as `Record<.....` above, we can also remove `const newObject: any`.

```

function mergeObjects(value: Record<string, unknown>, extraArgs: unknown[]): unknown {
	const [other] = extraArgs;

	if (!other) {
		return value;
	}

	if (typeof other !== 'object') {
		throw new ExpressionExtensionError('merge(): expected object arg');
	}

	const newObject = { ...value };
	for (const [key, val] of Object.entries(other)) {
		if (!(key in newObject)) {
			newObject[key] = val;
		}
	}
	return newObject;
}
```

We also have to update the assertions when this method is called from `object` to `Record<string, unknown>`

Reply:
Nice one! Updating.